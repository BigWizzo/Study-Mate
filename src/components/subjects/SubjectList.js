import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { subjectFetchRequest } from '../../redux/subjects/subjectActions';
import SubjectListItem from './SubjectListItem';

const SubjectList = ({ loadSubjects, subjects, history, student }) => {
  useEffect(() => {
    if (student) {
      loadSubjects();
      debugger;
    } else {
      history.push('./subjects');
    }
  }, [student]);

  const addSubject = () => {
    history.push('./subjects/new');
  };

  console.log(student);
  console.log(subjects);
  console.log(student);

  return (
    <div>
      <h1>Subject</h1>
      <div div className="row">
        {subjects?.length > 0 ? (
          subjects.map((subject) => (
            <div className="col-6 col-md-4">
              <SubjectListItem subject={subject} />
            </div>
          ))
        ) : (
          <div>Nothing to show yet</div>
        )}
        <button className="btn btn-primary" type="button" onClick={addSubject}>
          Add Subject
        </button>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => ({
  subjects: state.subjects.subject,
  student: state.student.student,
});

const mapDispatchToProps = (dispatch) => ({
  loadSubjects: () => dispatch(subjectFetchRequest()),
});

export default connect(mapStateToProps, mapDispatchToProps)(SubjectList);
