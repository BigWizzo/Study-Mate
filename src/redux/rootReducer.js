import { combineReducers } from 'redux';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import studentReducer from './students/studentReducer';
import subjectReducer from './subjects/subjectReducer';
import clockingReducer from './clockings/clockingReducer';

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['student', 'subjects', 'clockings'],
};

const rootReducer = combineReducers({
  student: studentReducer,
  subjects: subjectReducer,
  clockings: clockingReducer,
});

export default persistReducer(persistConfig, rootReducer);
