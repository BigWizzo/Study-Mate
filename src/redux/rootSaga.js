import { all, call } from 'redux-saga/effects';
import { studentSagas } from './students/studentSagas';
import { subjectSagas } from './subjects/subjectSagas';

export default function* rootSaga() {
  yield all([call(studentSagas), call(subjectSagas)]);
}
