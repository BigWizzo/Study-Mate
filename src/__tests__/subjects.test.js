import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import { store, persistor } from '../redux/store';
import SubjectList from '../components/subjects/SubjectList';

const subject = {
  id: 1,
  title: 'title 1',
  description: 'description 1',
};

test('should render the subject list component', () => {
  const component = renderer.create(
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <BrowserRouter>
          <SubjectList
            id={subject?.id}
            title={subject?.title}
            description={subject?.description}
          />
        </BrowserRouter>
      </PersistGate>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('<SubjectList />', () => {
  let component;
  beforeEach(() => {
    component = render(
      <Provider store={store}>
        <PersistGate loading={null} persistor={persistor}>
          <BrowserRouter>
            <SubjectList
              id={subject?.id}
              title={subject?.title}
              description={subject?.description}
            />
          </BrowserRouter>
        </PersistGate>
      </Provider>,
    );
  });

  test('Should have heading', () => {
    component.getByText('Subjects');
  });
});
