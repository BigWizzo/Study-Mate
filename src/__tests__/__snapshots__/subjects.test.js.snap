// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the subject list component 1`] = `
<div
  className="container-fliuid"
>
  <div
    className="row py-3 mb-2 bg-white"
  >
    <span
      className="col-2 text-center"
    >
      <i
        className="fas fa-angle-left"
      />
    </span>
    <span
      className="col-8 text-center"
    >
      21 September 2021
    </span>
    <span
      className="col-2 text-center"
    >
      <i
        className="fas fa-angle-right"
      />
    </span>
  </div>
  <h1
    className="text-center my-2"
  >
    Subjects
  </h1>
  <div
    className="row"
  >
    <div
      className="text-center"
    >
      Nothing to show yet. Please add some subjects
    </div>
  </div>
</div>
`;
